modules/indexer/stats/db.go

```diff
@@ -47,8 +51,62 @@ func (db *DBIndexer) Index(id int64) error {
 		return nil
 	}
 
+	var tmpIndex *os.File
+	if git.CheckGitVersionAtLeast("1.7.8") == nil {
+		tmpIndex, err = ioutil.TempFile("", "index")
+		if err != nil {
+			return err
+		}
+		defer func() {
+			err := util.Remove(tmpIndex.Name())
+			if err != nil {
+				log.Error("failed to remove tmp index file: %v", err)
+			}
+		}()
+
+		_, err = git.NewCommand("read-tree", commitID).
+			RunInDirWithEnv(gitRepo.Path, []string{"GIT_INDEX_FILE=" + tmpIndex.Name()})
+		if err != nil {
+			return err
+		}
+	}
+
 	// Calculate and save language statistics to database
-	stats, err := gitRepo.GetLanguageStats(commitID)
+	stats, err := gitRepo.GetLanguageStats(commitID, func(path string) (string, bool) {
+		// get language follow linguist rulers
+		// linguist-language=<lang> attribute to an language
+		// linguist-vendored attribute to vendor or un-vendor paths
+
+		if tmpIndex == nil {
+			return "", false
+		}
+
+		name2attribute2info, err := gitRepo.CheckAttribute(git.CheckAttributeOpts{
```

Comment:
OK Now I understand what this is doing - I think we're gonna need a pipe which goes to write to an open

```
git check-attr --stdin ...
```

and then read and parse the line produced per path. 

Otherwise you're going to get n calls to git check-attr per repository.